core/libp2p/muxer/connection_muxer.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CONNECTION_MUXER_HPP
+#define KAGOME_CONNECTION_MUXER_HPP
+
+#include <memory>
+
+#include "libp2p/stream/stream.hpp"
+#include "libp2p/transport/connection.hpp"
+#include "libp2p/transport/muxed_connection.hpp"
+
+namespace libp2p::muxer {
+  /**
+   * Strategy to upgrade connections to muxed
+   */
+  class StreamMuxer {
+   public:
+    using NewStreamHandler =
+        std::function<void(std::unique_ptr<stream::Stream>)>;
+
+//    /**
+//     * Upgrade connection to the muxed one
```

Comment:
What should be here instead of commented out code?

Reply:
Probably, the same function. But, if we decide the interfaces should be different, some another function(s) :)